File: trunk/uim/storage/mongo/src/test/java/eu/europeana/uim/store/mongo/MongoStorageTest.scala
package eu.europeana.uim.store.mongo

import com.mongodb.Mongo
import eu.europeana.uim.api.StorageEngine
import eu.europeana.uim.store.AbstractStorageEngineTest
import org.junit.Assert.fail
import scala.collection.JavaConversions._
import org.junit.runner.RunWith
import org.junit.runners.JUnit4


/**
 * @author Manuel Bernhardt <bernhardt.manuel@gmail.com>
 */
@RunWith(classOf[JUnit4])
object MongoStorageTest {
  var mongoEngine: MongoStorageEngine = null
  val m: Mongo = new Mongo
}

class MongoStorageTest extends AbstractStorageEngineTest {


  protected[store] def getStorageEngine:MongoStorageEngine = {
    if (MongoStorageTest.mongoEngine == null) {
      try {
        val engine: MongoStorageEngine = new MongoStorageEngine("UIMTEST")
        MongoStorageTest.m.dropDatabase("UIMTEST")
        engine.initialize
        MongoStorageTest.mongoEngine = engine
      }
      catch {
        case e: Exception => {
          e.printStackTrace
          fail("Could not initialize mongodb storage engine")
        }
      }
    }
    else {
      return MongoStorageTest.mongoEngine
    }
    return null

  }

  protected[store] override def performSetUp = {
    MongoStorageTest.m.getDB("UIMTEST").getCollectionNames() filter (name => name.startsWith("Mongo") || name.startsWith("records") ) foreach (name => MongoStorageTest.m.getDB("UIMTEST").getCollection(name).drop)
  }

  override def testCreateAndGetProvider = {
    // TODO figure out why the hell JUnit doesn't invoke @Before for the first method of the suite
    super.setUp;
    super.testCreateAndGetProvider
  }
}

File: trunk/uim/storage/mongo/src/test/java/eu/europeana/uim/store/mongo/MongoStorageTest.java
package eu.europeana.uim.store.mongo;

import java.net.UnknownHostException;

import org.junit.After;
import org.junit.Before;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import com.mongodb.Mongo;
import com.mongodb.MongoException;

import eu.europeana.uim.api.AbstractStorageEngineTest;
import eu.europeana.uim.api.StorageEngine;

/**
 * Configuration class for MongoDB StorageEngineTests
 * 
 * @author Georgios Markakis
 */
@RunWith(JUnit4.class)
public class MongoStorageTest extends AbstractStorageEngineTest<Long> {

	private MongoStorageEngine mongoEngine = null;

    private Mongo m = null;	
	
	/**
	 * Run before each test
	 */
	@Before
    public void setupTest(){
		
		try {
			m = new Mongo();
		} catch (UnknownHostException e) {
			e.printStackTrace();
		} catch (MongoException e) {
			e.printStackTrace();
		}
	}
    
	/**
	 * Run after each test
	 */
	@After
    public void cleanup(){
		m.dropDatabase("UIMTEST");
	}
	
	/* (non-Javadoc)
	 * @see eu.europeana.uim.store.AbstractStorageEngineTest#getStorageEngine()
	 */
	@Override
	protected StorageEngine<Long> getStorageEngine() {
		   if (mongoEngine == null) {
			      try {
			    	m = new Mongo();
			        MongoStorageEngine engine = new MongoStorageEngine("UIMTEST");
			        m.dropDatabase("UIMTEST");
			        engine.initialize();
			        mongoEngine = engine;
			      }
			      catch(Exception e) {
			          e.printStackTrace();
			      }
			    }
			    else {
			      return mongoEngine;
			    }
		   return mongoEngine;
	}
	
	
	

}
